### PR TITLE
Revamp the "Build" section.

### LINKED ISSUE #248: Improve the Build section.
A lot of the contents in the `/build` section is good, but most of it has been written by multiple writers, making it disjointed and not very friendly to look at. We should improve the Build section by:

- [ ] Adding a purpose for each page.
- [ ] Adding further reading/resources to each page, where necessary.
- [x] Re-structure the `/build/examples` into single pages for clarity (and to make the sidebar nav easier to deal with).
- [x] Grammar and spell check _everything_.
- [ ] List redirects needed after moving things around.

### PR DESCRIPTION
**Pull request by johnnymatthews**
The Build section is a bit of a mess, and needs reorganizing. This PR will likely need tweaking to fit with @cwaring's [layout and sidebar changes](https://github.com/filecoin-project/filecoin-docs/pull/239).

Closes #248 

## Tasks

- [ ] Add all the networks to the `build/networks` page.
- [ ] In `build/getting-started` explain how to _set up a node_ once the reader has finished installing Lotus (there likely isn't much they have to do).
- [ ] Add a simple _ping your node_ command on the `build/getting-started` page to test the readers set up.
- [ ] Include some common config suggestions for a dev node.
- [ ] Fix sub-sub-sub sections in tutorials (`5a xyz`, etc)

### CODE REVIEW

**Comment by cwaring** [comment]
Taking a quick look over this and there are a few issues I think we should solve before going this route.

1) You've broken many of the content links by changing from relative paths, all internal content links in the documents must be `../folder/markdown.md` instead of to `/absolute/path/` because otherwise we can't automatically check them for 404s and it's not possible to navigate from GithHub.

2) If make the tutorials into a single pages we need to consider a couple of things. Goal tracking in analytics will be much harder as we don't not if a user has reached the final step and we only get a single "was this helpful" vote for the whole tutorial. We may need to discuss what our targets are with these tutorials in order to pick the right layout approach OR build a custom view for really long pages so we can improve the UX and trigger correct metrics.

3) These sidebar changes will conflict once #239 has been merged so any changes there will need to manually ported across. Prepare for a lot of red lines 😅

**Comment by johnnymatthews** [comment]
> ... You've broken some of the content links by changing from relative paths ...

@cwaring No worries, I'll fix them now.

> ... If we make the tutorials into a single pages we need to consider a couple of things...

@cwaring if this is too hard to do right now then **let's just scrap it**. I found going through the tutorials in there current state confusing and convoluted, and I had to keep referring back to previous pages to understand what as going on. That problem is because of the general structure of these tutorials needs re-writing, but because we're short on time moving things into a single page was a quick fix. But again, if it's too hard to get the right analytics sorted, just scrap it and we'll re-organize the tutorials down the line.

> ... These sidebar changes will conflict ...

Yeahp, I'm aware. That's why this PR is still a draft. Waiting for that merge to go in so I can cherry pick the changes from here.

**Comment by cwaring** [comment]
@johnnymatthews thanks! I would rather us work towards the best reading experience and if you think a single page for these types of tutorials works then let's to that and I can wire in any extras to fit later, for now I just wanted to throw is into the ring for consideration :)

**Comment by mishmosh** [comment]
The `networks.md` page needs to stay outside of the Build section, since many different types of users (miners, builders, clients) all need to reference that info.

In general this is a MASSIVE PR that is hard to follow along with. Any chance this could be split into smaller changes — the refactor first, and then significant batches of content edits in their own PRs?

**Comment by johnnymatthews** [comment]
@cwaring I'm confused. Do you want me to revert the `examples` section to how they were (separate pages) or leave my changes in here?

> The networks.md page needs to stay outside of the Build section

@mishmosh yeah probably. Where do you reckon it should go? It could go into the new _Install Filecoin_ section ([pending PR review](https://github.com/filecoin-project/filecoin-docs/pull/235)).

> In general this is a MASSIVE PR that is hard to follow along with.

@mishmosh yeah sorry about that. I was told we needed to move fast and get stuff out the door before Space Race. I didn't expect this PR to face much review/discussion. If @cwaring would like to revert back to the `examples` section to how they were then I'll just close this PR, make a new one, and cherry pick in the small changes.

**Comment by cwaring** [comment]
> @cwaring I'm confused. Do you want me to revert the `examples` section to how they were (separate pages) or leave my changes in here?

Keep them as single pages if that improves the reading exp, I just wanted to note potential issues so we can track and fix in future updates. If we use this same pattern for other tutorials then that will be great.

File: docs/build/implementations.md [hunk 1]
```diff
@@ -29,11 +29,11 @@ Here is a snapshot of each implementation’s progress across the primary parts
 
 ✅ : fully featured implementation | 🔄 : reuses components from another implementation | 🔶 : partial implementation
```
Comment (brunolm) [inline docs/build/implementations.md:30]: 1 icon per line would be easier to read:

✅ : fully featured implementation
🔄 : reuses components from another implementation
🔶 : partial implementation

Comment (johnnymatthews) [reply docs/build/implementations.md:30]: Good point. Let's see if I can add that without playing around with GitHub's funky editor.

```suggestion
✅ : fully featured implementation
🔄 : reuses components from another implementation
🔶 : partial implementation
```

**Comment by johnnymatthews** [comment]
Closing this PR because it's become far too difficult to manage. I've created another, easier to follow, PR that contains a handful of the issues addressed here. #332.
